src/MetricDiffTrait.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Tightenco\NovaGoogleAnalytics;
+
+use Carbon\Carbon;
+
+trait MetricDiffTrait
+{
+    private function getPeriodDiff($startDate) {
```

Comment:
```suggestion
    private function getPeriodDiff($startDate)
    {
```

Reply:
Let's typehint startDate just for documentation

Reply:
I think this means:

Assuming today is the end of a period
and the provided startdate was the first day of that period
give me the start and end of the *previous* period

method name probably more like "getPreviousPeriodDates"

Maybe, in order to make this clearer:

```php
    private function getPreviousPeriodDates($thisPeriodStart, $thisPeriodEnd = null)
    {
        if (! $thisPeriodEnd) {
            $thisPeriodEnd = Carbon::today();
        }
        
        //
        
        return [$lastPeriodStart, $lastPeriodEnd];
    }
```

potential simpler code:

```php
$thisPeriodLength = $thisPeriodStart->diffInDays($thisPeriodEnd);
$lastPeriodEnd = $thisPeriodStart->clone()->subDay();
$lastPeriodStart = $lastPeriodEnd->clone()->subDays($thisPeriodLength);
```